docs/kore-syntax.md

```diff
@@ -1,296 +1,326 @@
-
-<sort-definition>
-  ::= "sort" <sort-id> "{" <sort-variables> "}"
-        "[" <attributes> "]"
-<hooked-sort-definition>
-  ::= "hooked-sort" <sort-id> "{" <sort-variables> "}"
-        "[" <attributes> "]"
-
-<symbol-definition>
-  ::= "symbol" <symbol> "(" <sorts> ")" ":" <sort>
-        "[" <attributes> "]"
-<hooked-symbol-definition>
-  ::= "hooked-symbol" <symbol> "(" <sorts> ")" ":" <sort>
-        "[" <attributes> "]"
-
-<alias-definition>
-  ::= "alias" <alias> "(" <sorts> ")" ":" <sort>
-        "where" <application-pattern> ":=" <pattern>
-        "[" <attributes> "]"
-
-<axiom>
-  ::= "axiom" "{" <sort-variables> "}"
-        <pattern>
-        "[" <attributes> "]"
-<claim>
-  ::= "claim" "{" <sort-variables> "}"
-        <pattern>
-        "[" <attributes> "]"
+<keyword>
+  ::= "module"  | "endmodule"
+    | "import"
+    | "sort"    | "hooked-sort"
+    | "symbol"  | "hooked-symbol"
+    | "axiom"   | "claim"
+    | "alias"   | "where"
+```
 
-<alias> ::= <symbol-or-alias>
-<symbol> ::= <symbol-or-alias>
-<symbol-or-alias> ::= <symbol-id> "{" <sort-variables> "}"
+### Identifiers
 
-<sort-variable>
-  ::= <sort-id>
+```
+<identifier>
+  ::= <id-first-char> <id-char>*
+
+<identifier-char>
+  ::= <identifier-first-char>
+    | <identifier-other-char>
 
+<identifier-first-char>
+  ::= ['A'..'Z', 'a'..'z']
+
+<identifier-other-char>
+  ::= ['0'..'9', '\'', '-']
+
+<symbol-identifier>
+  ::= ['\']? <identifier>
+
+<element-variable-identifier>
+  ::= <identifier>
+
+<set-variable-identifier>
+  ::= ['@'] <identifier>
+
+<sort-identifier>
+  ::= <identifier>
+
+<module-name>
+  ::= <identifier>
+```
+
+Identifiers must not be `<keyword>`s.
+
+## Syntax
+
+The syntax of Kore is defined here in [Backus-Naur form](https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form),
+augmented with the following meta-syntactic notations for brevity:
+
+```
+<[$item]-sep-by-[$separator]>
+  ::= ""
+    | <[$item]-sep-by-1-[$separator]>
+
+<[$item]-sep-by-1-[$separator]>
+  ::= $item
+    | $item $separator <[$item]-sep-by-1-[$separator]>
+```
+
+### Sorts
+
+A sort is either a _sort variable_ or a _sort constructor_ applied to a list of _sort parameters_.
+
+```
 <sort>
   ::= <sort-variable>
-    | <sort-id> "{" <sorts> "}"
+    | <sort-identifier> "{" <sorts> "}"
+
+<sorts>
+  ::= <[sort]-sep-by-[","]>
+
+<sort-variable>
+  ::= <sort-identifier>
+
+<sort-variables>
+  ::= <[sort-variable]-sep-by-[","]>
 ```
 
 ### Patterns
 
 ```
 <pattern>
-  ::= <element-variable>
-    | <set-variable>
-    | <ML-pattern>
+  ::= <variable-pattern>
+    | <matching-logic-pattern>
     | <application-pattern>
     | <string-literal>
 
-<variable>
+<patterns>
+  ::= <[pattern]-sep-by-[","]>
+
+<variable-pattern>
   ::= <element-variable>
     | <set-variable>
+
 <element-variable>
-  ::= <element-variable-id> ":" <sort>
+  ::= <element-variable-identifier> ":" <sort>
+
 <set-variable>
-  ::= <set-variable-id> ":" <sort>
+  ::= <set-variable-identifier> ":" <sort>
```

Comment:
Aren't set variables required to be prefixed with `@`?

Reply:
That's part of `<set-variable-identifier>`.